solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/batch/SolrSearchSink.java

```diff
@@ -0,0 +1,146 @@
+  private final BatchSolrSearchConfig batchConfig;
+
+  public SolrSearchSink(BatchSolrSearchConfig batchConfig) {
+    this.batchConfig = batchConfig;
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    context.addOutput(Output.of(batchConfig.referenceName, new SolrSearchSink.SolrOutputFormatProvider(batchConfig)));
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    batchConfig.validateSolrConnectionString();
+    if (inputSchema != null) {
+      batchConfig.validateKeyField(inputSchema);
+      batchConfig.validateInputFieldsDataType(inputSchema);
+    }
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    batchConfig.verifySolrConfiguration();
+  }
+
+  @Override
+  public void transform(StructuredRecord structuredRecord, Emitter<KeyValue<Text, Text>> emitter) throws Exception {
+    batchConfig.validateKeyField(structuredRecord.getSchema());
+    batchConfig.validateInputFieldsDataType(structuredRecord.getSchema());
+
+    if (structuredRecord.get(batchConfig.getKeyField()) == null) {
+      return;
+    }
+    emitter.emit(new KeyValue<Text, Text>(new Text(GSON.toJson(structuredRecord.getSchema(), SCHEMA_TYPE)),
+                                          new Text(StructuredRecordStringConverter.toJsonString(structuredRecord))));
+  }
+
+  /**
+   * Output format provider for Batch SolrSearch Sink.
+   */
+  private static class SolrOutputFormatProvider implements OutputFormatProvider {
+    private Map<String, String> conf;
+
+    SolrOutputFormatProvider(BatchSolrSearchConfig batchConfig) {
+      this.conf = new HashMap<>();
+      conf.put(SolrRecordWriter.SERVER_URL, batchConfig.getSolrHost());
+      conf.put(SolrRecordWriter.SERVER_MODE, batchConfig.getSolrMode());
+      conf.put(SolrRecordWriter.COLLECTION_NAME, batchConfig.getCollectionName());
+      conf.put(SolrRecordWriter.KEY_FIELD, batchConfig.getKeyField());
+      conf.put(SolrRecordWriter.BATCH_SIZE, batchConfig.getBatchSize());
+      if (batchConfig.getOutputFieldMappings() == null) {
+        conf.put(SolrRecordWriter.FIELD_MAPPINGS, "");
+      } else {
+        conf.put(SolrRecordWriter.FIELD_MAPPINGS, batchConfig.getOutputFieldMappings());
+      }
+    }
+
+    @Override
+    public String getOutputFormatClassName() {
+      return SolrOutputFormat.class.getName();
+    }
+
+    @Override
+    public Map<String, String> getOutputFormatConfiguration() {
+      return conf;
+    }
+  }
+
+  /**
+   * Config class for Batch SolrSearch sink.
+   */
+  public static class BatchSolrSearchConfig extends SolrSearchSinkConfig {
+    @Description("Number of documents to create a batch and send it to Solr for indexing. After each batch, " +
+      "commit will be triggered. Default batch size is 10000.")
+    private final String batchSize;
```

Comment:
you should enable macros for these fields. maybe in the next PR. but we should consider that when writing plugins.